ui/src/components/admin/sermonNotes/SermonNotesEditorModal.js

```diff
@@ -0,0 +1,384 @@
+import React, { useEffect } from 'react';
+import {
+  Heading,
+  Container,
+  FormControl,
+  Box,
+  FormLabel,
+  Input,
+  FormHelperText,
+  Grid,
+  Stack,
+  Button,
+  useToast,
+  Select,
+} from '@chakra-ui/react';
+import { set, useForm } from 'react-hook-form';
+import { useState } from 'react';
+import { customAxios as axios } from '../../helpers/customAxios';
+
+const SermonNotesEditorModal = (props) => {
+  const { user, editSermonNotesData, actionOnEditor, setIsEditorOpen } = props;
+  const {
+    register,
+    handleSubmit,
+    reset,
+    setValue,
+    formState: { errors },
+  } = useForm();
+  const toast = useToast();
+
+  // Create a object
+  const sermonIdMap = {
+    'Sunday Celebration': 'sn',
+    Encounter: 'encounter',
+  };
+
+  const [title, setTitle] = useState(undefined);
+  const [sermondId, setSermonId] = useState(undefined);
+  const [subtitle, setSubtitle] = useState(undefined);
+  const [speaker, setSpeaker] = useState(undefined);
+  const [sermonSeries, setSermonSeries] = useState(undefined);
+  const [imageLink, setImageLink] = useState(undefined);
+  const [originalContent, setOriginalContent] = useState(undefined);
+  const [sermonLink, setSermonLink] = useState(undefined);
+  const [serviceType, setServiceType] = useState(undefined);
+  const [passage, setPassage] = useState(undefined);
+  const [date, setDate] = useState(undefined);
+
+  // This is the state that will hold the number of sermons for a particular date for edge case handling
+  const [numberOfSermons, setNumberOfSermons] = useState(0);
+
+  const editorSubmitButton = (actionOnEditor) => {
+    switch (actionOnEditor) {
+      default:
+        return 'Create Sermon Notes';
+      case 'edit':
+        return 'Update Sermon Notes';
+      case 'duplicate':
+        return 'Duplicate Sermon Notes';
+    }
+  };
+
+  const setSermonNotesData = (data) => {
+    if (data) setValue('title', data.title);
+    setValue('subtitle', data.subtitle);
+    setValue('speaker', data.speaker);
+    setValue('sermonSeries', data.sermonSeries);
+    setValue('date', data.date);
+    setValue('imageLink', data.imageLink);
+    setValue('originalContent', data.originalContent);
+    setValue('sermonLink', data.sermonLink);
+    setValue('serviceType', data.serviceType);
+    setValue('passage', data.passage);
+
+    setTitle(data.title);
+    setSubtitle(data.subtitle);
+    setSpeaker(data.speaker);
+    setSermonSeries(data.sermonSeries);
+    setDate(data.date);
+    setImageLink(data.imageLink);
+    setOriginalContent(data.originalContent);
+    setSermonLink(data.sermonLink);
+    setServiceType(data.serviceType);
+    setPassage(data.passage);
+  };
```

Comment:
another way to make it cleaner, you can define an object in a react useState so u dont have to do this one by one.
```
const [sermonNotesData, setSermonNotesData] = useState({
title: '',
subtitle:'',
... the rest of the object fields ...
})
```
This way, you dont need to have all the states above too.

Reply:
Good Idea, will try to remake this